interface/src/Application.cpp

```diff
@@ -3182,13 +3182,37 @@ static const QUrl AUTHORIZED_EXTERNAL_QML_SOURCE { "https://content.highfidelity
 
 void Application::initializeUi() {
 
+    
     // Allow remote QML content from trusted sources ONLY
     {
         auto defaultUrlValidator = OffscreenQmlSurface::getUrlValidator();
-        auto newValidator = [=](const QUrl& url)->bool {
-            if (AUTHORIZED_EXTERNAL_QML_SOURCE.isParentOf(url)) {
-                return true;
+        auto newValidator = [=](const QUrl& url) -> bool {
+            QString whitelistPrefix = "[WHITELIST ENTITY SCRIPTS]";
+            QList<QString> safeURLS = { "" };
+            safeURLS += qEnvironmentVariable("EXTRA_WHITELIST").trimmed().split(QRegExp("\\s*,\\s*"), QString::SkipEmptyParts);
+
+            // PULL SAFEURLS FROM INTERFACE.JSON Settings
+
+            QVariant raw = Setting::Handle<QVariant>("private/settingsSafeURLS").get();
+            QStringList settingsSafeURLS = raw.toString().trimmed().split(QRegExp("\\s*[,\r\n]+\\s*"), QString::SkipEmptyParts);
+            safeURLS += settingsSafeURLS;
+
+            // END PULL SAFEURLS FROM INTERFACE.JSON Settings
+
+            bool isInWhitelist = false;  // assume unsafe
+            for (const auto& str : safeURLS) {
+                qDebug() << "url.toString().startsWith(str) = " << url.toString().startsWith(str);
+                qDebug() << "str.endsWith('.qml ') = " << str.endsWith(".qml");
```

Comment:
if you keep the debug lines here it would be preferred to use qCDebug instead and assign the right category, otherwise it will be hard to find or filter in the log.

Reply:
Like so:
`qCDebug(scriptengine) << `
This is how I did it for the Entity Script Whitelist but I removed the debug statements, though I must say it might still be necessary to at least post if it's been approved since it's easier to do that code-wise, as if you don't see an approval then you should assume it failed (When debugging as an end-user.)

Reply:
dude I have no idea what qCDebug is

Reply:
A variant on qDebug which also takes the category. See https://doc.qt.io/qt-5/qloggingcategory.html#qCDebug-1     .  There are many categories already defined see if you find one that is appropriate